src/core/models/utils.cc

```diff
@@ -23,37 +23,58 @@
 #include "models/utils.h"
 #include "parallel/api.h"
 
+
+/**
+ *  Generate a random multiplier and increment for quasi-random generator
+ *  based on the modular arithmetics. When seed is zero, we just return
+ *  the default parameters: multiplier == 1 and increment == 0.
+ */
+ModularParams modular_random_gen(size_t n, unsigned int seed) {
+  ModularParams mp;
+  if (seed) {
+    auto multipliers = calculate_coprimes(n);
```

Comment:
Isn't it a bit excessive, to calculate all coprimes here? For example, if `n` is prime, then all numbers `1 .. n-1` are coprimes. 
A simpler strategy could be to just pick a number below n at random, check if it's coprime, and retry if not. After all, [the probability for two numbers to be coprime](https://math.stackexchange.com/questions/64498) is 6/pi^2 ≈ 60%.